integration_tests/test_quick.py

```diff
@@ -763,6 +766,72 @@ def test_repay_and_delay_interest(self):
         self.assertEqual(len(txs), 1)
         self.assertEqual(txs[0]["amount"], 13_000)      
         
+    def test_withraw_admin_rewards(self):
+        chain = LocalChain(storage=self.storage)
+
+        self.add_token(chain, token_a)
+        self.add_token(chain, token_b)
+
+        chain.execute(self.ct.mint(0, 40_000))
+        chain.execute(self.ct.enterMarket(0))
+        
+        chain.execute(self.ct.borrow(1, 10_000))
+
+        chain.advance_blocks(1)
+
+        self.update_price_and_interest(chain, 0, 100, one_percent_per_second)
+        self.update_price_and_interest(chain, 1, 100, one_percent_per_second)
+
+        chain.execute(self.ct.repay(1, 13_000))
+        res = chain.execute(self.ct.withdrawReserve(1, 1_500), sender=admin)
+        transfers = parse_transfers(res)
+        self.assertEqual(transfers[0]["amount"], 1_500)
+
+        # nothing left to repay and withdraw
+        with self.assertRaises(MichelsonRuntimeError):
+            chain.execute(self.ct.repay(1, 1))
+            chain.execute(self.ct.withdrawReserve(1, 1), sender=admin)
+
+        # bob borrows [0] in the meantime
+        chain.execute(self.ct.mint(1, 40_000), sender=bob)
+        chain.execute(self.ct.enterMarket(1), sender=bob)
+        chain.execute(self.ct.borrow(0, 10_000), sender=bob)
+
+        chain.advance_blocks(1)
+
+        self.update_price_and_interest(chain, 0, 100, one_percent_per_second)
+        self.update_price_and_interest(chain, 1, 100, one_percent_per_second)
+
+        chain.execute(self.ct.repay(0, 13_000), sender=bob)
+        # receive rewards after bob borrow
+        res = chain.execute(self.ct.withdrawReserve(0, 1_500), sender=admin)
+        transfers = parse_transfers(res)
+        self.assertEqual(transfers[0]["amount"], 1_500)
+        with self.assertRaises(MichelsonRuntimeError):
+            chain.execute(self.ct.withdrawReserve(0, 1), sender=admin)
+
+        # `me` repeats everything as the first time
+        chain.execute(self.ct.borrow(1, 10_000))
+
+        chain.advance_blocks(1)
+
+        self.update_price_and_interest(chain, 0, 100, one_percent_per_second)
+        self.update_price_and_interest(chain, 1, 100, one_percent_per_second)
+
+        # not meaningful, just to add some mess
+        chain.execute(self.ct.redeem(0, 10_000))
+        res = chain.execute(self.ct.repay(1, 0))
+        txs = parse_transfers(res)
+        self.assertEqual(len(txs), 1)
+        self.assertEqual(txs[0]["amount"], 13_000)
+        # receive rewards after alice borrow
+        res = chain.execute(self.ct.withdrawReserve(1, 1_500), sender=admin)
+        transfers = parse_transfers(res)
+        self.assertEqual(transfers[0]["amount"], 1_500)
+        # nothing left to repay and withdraw
+        with self.assertRaises(MichelsonRuntimeError):
+            chain.execute(self.ct.repay(1, 1))
+            chain.execute(self.ct.withdrawReserve(1, 1), sender=admin)
```

Comment:
Please move this to a separate `with self.assertRaises`

Reply:
Resolved in [2a7774e104ceaf981498e039fc33fde0bc51a111](https://github.com/madfish-solutions/yupana-protocol-core/pull/75/commits/2a7774e104ceaf981498e039fc33fde0bc51a111)